source/common/event/dispatched_thread.cc

```diff
@@ -0,0 +1,50 @@
+#include "common/event/dispatched_thread.h"
+
+#include <chrono>
+
+#include "envoy/event/dispatcher.h"
+#include "envoy/event/timer.h"
+#include "envoy/server/configuration.h"
+
+#include "common/api/api_impl.h"
+#include "common/common/thread.h"
+
+#include "spdlog/spdlog.h"
+
+namespace Envoy {
+namespace Event {
+
+void DispatchedThreadImpl::start(Server::GuardDog& guard_dog) {
+  // Silly trick for preventing the event loop from exiting when there are no
+  // events: keep a long duration timer going in perpetuity.
+  auto no_exit = [this]() -> void {
+    no_exit_timer_->enableTimer(
+        std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::hours(1)));
```

Comment:
Agreed it would be nice to make worker use this class if possible to avoid this kind of duplication (and not too annoying of a refactor). If the refactoring is too annoying you might still be able to create a small mix-in utility class that could have the no exit timer, etc.